sklearn/naive_bayes.py

```diff
@@ -1528,3 +1541,600 @@ def _joint_log_likelihood(self, X):
+
+        Callable columns are replaced with sets of actual str or int indices.
+        Estimators that don't require fitting are passed as they were,
+        without cloning.
+        """
+        estimators_ = []
+        fitted_estimators = iter(fitted_estimators)
+
+        for name, nb_estimator, cols in self._iter():
+            if not _is_empty_column_selection(cols):
+                updated_nb_estimator = next(fitted_estimators)
+            else:  # don't advance fitted_estimators; use original
+                updated_nb_estimator = nb_estimator
+            estimators_.append((name, updated_nb_estimator, cols))
+        self.estimators_ = estimators_
+
+    def fit(self, X, y, sample_weight=None):
+        """Fit the naive Bayes meta-estimator.
+
+        Calls `fit` of each subestimator ``nb_estimator``.  Only a corresponding
+        subset of columns of `X` is passed to each subestimator; `sample_weight`
+        and `y` are passed to the subestimators as they are.
+
+        Parameters
+        ----------
+        X : array-like of shape (n_samples, n_features)
+            Training vectors, where `n_samples` is the number of samples
+            and `n_features` is the number of features.
+        y : array-like of shape (n_samples,)
+            Target values.
+        sample_weight : array-like of shape (n_samples,), default=None
+            Weights applied to individual samples (1. for unweighted).
+
+        Returns
+        -------
+        self : object
+            Returns the instance itself.
+        """
+        self._validate_params()
+        self._check_feature_names(X, reset=True)
+        # TODO: Consider overriding BaseEstimator._check_feature_names
+        # Currently, when X has all str feature names, all features are
+        # registered in self.feature_names_in no matter if they are used or not.
+        self._check_n_features(X, reset=True)
+        self._validate_estimators()
+        self._validate_column_callables(X)
+        # Consistency checks for X, y are delegated to subestimators
+
+        # Subestimators get original sample_weight. This is for class counts:
+        if sample_weight is not None:
+            weights = _check_sample_weight(sample_weight, X=y, copy=True)
+
+        # We would use sklearn.utils.multiclass.class_distribution, but it does
+        # not return class_count, which we want as well.
+        if sample_weight is None:
+            self.classes_, self.class_count_ = np.unique(
+                column_or_1d(y), return_counts=True
+            )
+        else:
+            self.classes_ = np.unique(column_or_1d(y))
+            counts = np.zeros(len(self.classes_))
+            for i, c in enumerate(self.classes_):
+                counts[i] = (weights * (column_or_1d(y) == c)).sum()
+            self.class_count_ = counts
+        self.n_classes_ = len(self.classes_)
+
+        estimators = list(self._iter(fitted=False, replace_strings=True))
```

Comment:
I need the length of the list for the log message. It can be shorter that `nb_estimators` passed at init, because some sub-estimators my have non-empty column sets. I can calculated the number at validation, but I don't think reducing the overhead of list creation justifies that. Or does it?